ui-service/src/main/frontend/hotel-booking-ui/src/components/search-components/RoomSelectorComponent.tsx

```diff
@@ -0,0 +1,161 @@
+import React, {useState, useRef, useEffect} from 'react';
+import {User, Minus, Plus, BedDouble, ChevronDown} from 'lucide-react';
+import './styles/RoomSelector.css';
+
+interface SearchHotelComponentProps {
+    rooms: number;
+    adults: number;
+    children: number;
+    onRoomDataChange: (key: 'rooms' | 'adults' | 'children', value: number) => void;
+}
+
+const SearchHotelComponent: React.FC<SearchHotelComponentProps> = ({rooms, adults, children, onRoomDataChange}) => {
```

Comment:
In the future, we should not use children (often used by react node prop) for local variables for better readability. It is clear at the top of the file where the type is defined, but I was confused by this many times towards the bottom.